ciao-controller/instance.go

```diff
@@ -46,7 +46,7 @@ type instance struct {
 	startTime time.Time
 }
 
-func isCNCIWorkload(workload *types.Workload) bool {
+func isCNCIWorkload(workload types.Workload) bool {
```

Comment:
> they told us to avoid passing by pointer if possible due to it complicating garbage collection and go's escape analysis. At what point do we need to worry about that?

I haven't seen this advice given elsewhere and it sounds a little bit like premature optimisation to me.  I think we only need to worry about this if we see that our services are doing too much GC.  This is probably something we should profile at some point.  The easiest way to get an idea is to prefix your executable with

GODEBUG=gctrace=1

Interestingly you can see which objects escape a function by passing using 

go build -gcflags="-m"